tests/test_ssl_transport_close.py

```diff
@@ -0,0 +1,141 @@
+import asyncio
+import platform
+import ssl
+import sys
+import warnings
+from typing import Any, Tuple, cast
+
+import pytest
+
+from aiohttp import TCPConnector, web
+from aiohttp.client import ServerDisconnectedError
+from aiohttp.pytest_plugin import AiohttpClient, AiohttpServer
+from aiohttp.test_utils import TestClient, TestServer
+
+
+async def handle_root(request: web.Request) -> web.Response:
+    """respond after server was restarted, forcing connection_lost"""
+    cq = request.app["cq"]
+    dq = request.app["dq"]
+
+    await dq.put(0)
+
+    await cq.get()
+    cq.task_done()
+
+    return web.Response(text="")
+
+
+async def _prepare(
+    aiohttp_server: Any,
+    aiohttp_client: Any,
+    ssl_ctx: ssl.SSLContext,
+    client_ssl_ctx: ssl.SSLContext,
+    cq: "asyncio.Queue[int]",
+    dq: "asyncio.Queue[int]",
+) -> Tuple[TestServer, TestClient]:
+    app = web.Application()
+    app["cq"] = cq
+    app["dq"] = dq
+    app.router.add_get("/", handle_root)
+    server = await aiohttp_server(app, port=0, ssl=ssl_ctx)
+
+    session = await aiohttp_client(server, connector=TCPConnector(ssl=client_ssl_ctx))
+
+    # replace SockSite … it is different and blocks
+    await dq.put(0)
+    await _restart(server.runner, ssl_ctx, session.port, cq, dq)
+    await cq.get()
+
+    return server, session
+
+
+async def _restart(
+    runner: web.BaseRunner,
+    ssl_ctx: ssl.SSLContext,
+    port: int,
+    cq: "asyncio.Queue[int]",
+    dq: "asyncio.Queue[int]",
+) -> None:
+    """restart service to force connection_lost"""
+    await dq.get()
+    dq.task_done()
+    for s in list(runner.sites):
+        await s.stop()
+        await cq.put(0)
+        s = web.TCPSite(
+            runner, "127.0.0.1", port=port, ssl_context=ssl_ctx, shutdown_timeout=1.0
+        )
+        await s.start()
+        break
```

Comment:
Just noticed this one in the codecov.
The break means this never iterates, so there's no point of a loop here.

Reply:
If the intention is just to use any site from the set, then try `site = next(iter(runner.sites))`.

Reply:
We only have 1 site.

`site = next(iter(runner.sites))` would be limited to 1 site as well?

Reply:
Yes, my point is that, if we are not iterating, then we don't need a loop. Which will also clear up coverage, which highlights the for loop as a partial, because it only runs once.